view/src/main/java/cmm/apps/esmorga/view/MainActivity.kt

```diff
@@ -45,9 +63,51 @@ class MainActivity : ComponentActivity() {
 
     private fun setupNavigation(loggedIn: Boolean) {
         setContent {
-            val navigationController = rememberNavController()
-            EsmorgaNavigationGraph(navigationController = navigationController, loggedIn = loggedIn)
+            EsmorgaTheme {
+                val navigationController = rememberNavController()
+                val bottomNavItems = listOf(
+                    BottomNavItem.Explore,
+                    BottomNavItem.MyEvents,
+                    BottomNavItem.Profile
+                )
+
+                HomeView(bottomNavItems, navigationController) {
+                    EsmorgaNavigationGraph(navigationController = navigationController, loggedIn)
+                }
+            }
+        }
+    }
+}
+
+@SuppressLint("UnusedMaterial3ScaffoldPaddingParameter")
+@Composable
+fun HomeView(bottomNavItems: List<BottomNavItem>, navigationController: NavHostController, content: @Composable () -> Unit) {
+    Scaffold(
+        modifier = Modifier.fillMaxSize(),
+        bottomBar = {
+            val navBackStackEntry by navigationController.currentBackStackEntryAsState()
+            val currentRoute =
+                navBackStackEntry?.destination?.hierarchy?.first()?.route?.substringAfterLast(".")
+            val items = bottomNavItems.map {
+                it.route.toString()
+            }
+
+            val visibility = currentRoute in items
+            HomeBottomBar(bottomNavItems, visibility, navigationController, currentRoute)
         }
+    ) {
+        content.invoke()
     }
+}
 
+@Composable
+fun HomeBottomBar(bottomNavItems: List<BottomNavItem>, visibility: Boolean, navigationController: NavHostController, currentRoute: String?) {
```

Comment:
can `currentRoute` be a typed object instead of a String? Latest Navigation version includes some types already, may them be used here too?